### PR TITLE
Add note to 2.1.4 Character Key Shortcuts understanding

### PR DESCRIPTION
**Pull request by patrickhlauke**
Make the point that while the SC itself refers to "character keys", it's nothing to do with how many keys are pressed.
Also corrects a typo, gives the document a proper `<title>` (though irrelevant, as this is changed on publication, it's still nice for good housekeeping), and uses `<kbd>` for the characters mentioned.

Closes https://github.com/w3c/wcag/issues/2314

### CODE REVIEW

**Review by bruce-usab — COMMENTED** [review]
Looks good!

**Comment by patrickhlauke** [comment]
@alastc any chance this could be discussed/considered at some point soon?

**Review by detlevhfischer — APPROVED** [review]
Should the revised understanding have something on videos processing single printable key shortcuts once they receive focus? The question has been popping up (can't remember where) whether the whole of the video as it receives focus can count as "user interface component" to allow it to be "active only on focus". I can see a good case for allowing that, and I'am not aware of drawbacks. Not sure whether this question should be lumped together with this clarifying rewrite though...

**Comment by patrickhlauke** [comment]
@detlevhfischer i'd keep that thing about video players separate, as a resolution to https://github.com/w3c/wcag/issues/1950

File: understanding/21/character-key-shortcuts.html [hunk 1]
```diff
@@ -15,58 +15,57 @@ <h2>In brief</h2>
...
-<p>A user interacting with a webpage or web app that doesn't use single-character shortcuts doesn't have this problem. Inadvertent strings of characters from the speech application are not interpreted as shortcuts if a modifier key is required. A speech user filling in a text input form may find that a phrase that is accidentally picked up by the speech microphone results in stray text being entered into the field, but that is easily seen and undone. The Resources section of this page contains links to videos demonstrating these types of issues.</p>
+      <p>The intent of this Success Criterion is to reduce accidental activation of keyboard shortcuts. Character key shortcuts work well for many keyboard users, but are inappropriate and frustrating for speech input users – whose means of input is strings of letters – and for keyboard users who are prone to accidentally hit keys. To rectify this issue, authors need to allow users to turn off or reconfigure shortcuts that are made up of only character keys.</p>
```
Comment (mbgower) [inline understanding/21/character-key-shortcuts.html:22]:
```suggestion
      <p>The intent of this Success Criterion is to reduce accidental activation of keyboard shortcuts. Character key shortcuts work well for many keyboard users, but are inappropriate and frustrating for speech input users, whose dictation is interpreted as strings of letters, and for keyboard users, who are prone to accidentally hit keys. To rectify this issue, authors need to allow users to turn off or reconfigure shortcuts that are made up of only character keys.</p>
```

File: understanding/21/character-key-shortcuts.html [hunk 2]
```diff
@@ -15,58 +15,57 @@ <h2>In brief</h2>
...
+			<div class="note">
+				<p>Even though this Success Criterion refers to <q>character keys</q>, note that it's not relevant whether a shortcut can be activated using a single physical key on a keyboard, or if it requires a combination of keys to be pressed. For instance, on most full-size US and UK keyboard, the <kbd>?</kbd> (question mark) symbol is accessed using <kbd>Shift</kbd>+<kbd>/</kbd> (forward slash key next to the right-hand <kbd>Shift</kbd> key). The <kbd>é</kbd> (lowercase e with an acute accent) requires the use of <kbd>AltGr</kbd>+<kbd>e</kbd>. The specific key combination required for certain characters will also vary depending on the user's keyboard layout. However, shortcuts that use these characters still fall under the requirements of this Success Criterion. What matters is that a shortcut relies on a printable character (letters, punctuation, numbers, symbol characters), and not the number of physical keyboard keys that users need to press to trigger it.</p>
```
Comment (fstrr) [inline understanding/21/character-key-shortcuts.html:24]: I think you said last week that you'd note that the AltGr example was specific to Windows. Also, if you're interested, you can nest `kbd` elements when you're referencing multiple keys: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-kbd-element

Comment (patrickhlauke) [reply understanding/21/character-key-shortcuts.html:24]: Done, and done

**Review by bruce-usab — APPROVED** [review]
Still looks good to me!

File: understanding/21/character-key-shortcuts.html [hunk 3]
```diff
@@ -15,62 +15,61 @@ <h2>In brief</h2>
...
-<p>A user interacting with a webpage or web app that doesn't use single-character shortcuts doesn't have this problem. Inadvertent strings of characters from the speech application are not interpreted as shortcuts if a modifier key is required. A speech user filling in a text input form may find that a phrase that is accidentally picked up by the speech microphone results in stray text being entered into the field, but that is easily seen and undone. The Resources section of this page contains links to videos demonstrating these types of issues.</p>
+      <p>The intent of this Success Criterion is to reduce accidental activation of keyboard shortcuts. Character key shortcuts work well for many keyboard users, but are inappropriate and frustrating for speech input users, whose dictation is interpreted as strings of letters, and for keyboard users, who are prone to accidentally hit keys. To rectify this issue, authors need to allow users to turn off or reconfigure shortcuts that are made up of only character keys.</p>
```
Comment (alastc) [inline understanding/21/character-key-shortcuts.html:22]:
```suggestion
      <p>The intent of this Success Criterion is to reduce accidental activation of keyboard shortcuts. Character key shortcuts work well for many keyboard users. However, they can be inappropriate and frustrating for speech input users whose dictation is interpreted as strings of letters, and for keyboard users who are prone to accidentally hit keys. To rectify this issue, authors need to allow users to turn off or reconfigure shortcuts that are made up of only character keys.</p>
```

Comment (alastc) [reply understanding/21/character-key-shortcuts.html:22]: Trying to reduce sentence complexity.

Comment (patrickhlauke) [reply understanding/21/character-key-shortcuts.html:22]: sure. i also like how this makes it less of an absolute statement and tempers it a touch.

**Comment by patrickhlauke** [comment]
@detlevhfischer sorry, a super-late rejoined to this: in a11y slack https://web-a11y.slack.com/archives/C7KTV5CBA/p1736926950552289 somebody pointed out that http://3needs.org/en/testing/code/kb-shortcuts.html seems to omit/not test all printable characters, such as `é` - any chance it could be modified/amended

**Comment by detlevhfischer** [comment]
@patrickhlauke Thanks, I'll pass this on to Telekom MMS, this is not our Bookmarklet I just used it to show its operation on a test page. This thing is that on a German (and presumably also English) keyboard, producing é actually involves hitting two keys so I am not sure if these special characters would really need to be covered - where do you stop? Other languages have other diacritical characters...

**Comment by patrickhlauke** [comment]
> This thing is that on a German (and presumably also English) keyboard, producing é actually involves hitting two keys

well that's the point of this PR here. it's not about "how many physical keys do you need to press", but rather "is it a printable character". and yes, in theory, you need to test the *entire* unicode ranges of printable characters...

**Comment by besenwagen** [comment]
> where do you stop?

In the context of printable [Latin-1 Supplement](https://en.wikipedia.org/wiki/Latin-1_Supplement) characters, 255 seems a much better choice than 191. 🤷‍♀️
